server/src/main/kotlin/com/trib3/server/graphql/GraphQLWebSocket.kt

```diff
@@ -24,13 +118,22 @@ private val log = KotlinLogging.logger { }
  */
 open class GraphQLWebSocket(
     val graphQL: GraphQL?,
-    val objectMapper: ObjectMapper
+    val objectMapper: ObjectMapper,
+    val scheduler: Scheduler = Schedulers.io() // use the io scheduler by default
 ) : WebSocketAdapter() {
-    val objectWriter = objectMapper.writerWithDefaultPrettyPrinter()
+    val objectWriter = objectMapper.writerWithDefaultPrettyPrinter()!!
+
+    private val remoteLock = ReentrantLock()
```

Comment:
what's the remote lock for?

Reply:
the jetty websocket remote object can only be called from one thread at a time, so this protects the `sendMessage` from failing with an obscure error if we send a keepalive and a data message at the same time: `Blocking message pending 10000 for BLOCKING`